src/main/c/org_apache_activemq_artemis_nativo_jlibaio_LibaioContext.c

```diff
@@ -32,15 +32,11 @@
 #include <fcntl.h>
 #include <stdlib.h>
 #include <pthread.h>
+#include <stdatomic.h>
 #include <limits.h>
 #include "org_apache_activemq_artemis_nativo_jlibaio_LibaioContext.h"
 #include "exception_helper.h"
 
-//x86 has a strong memory model and there is no need of HW fences if just Write-Back (WB) memory is used
-#define mem_barrier() __asm__ __volatile__ ("":::"memory")
-#define read_barrier()	__asm__ __volatile__("":::"memory")
-#define store_barrier()	__asm__ __volatile__("":::"memory")
-
```

Comment:
I think you might run into problems with a few older GCC versions here. I believe stdatomic requires GCC 5 or newer. You may work-around this with a few cmake checks and fall back to the older format in the worst scenario. This should be simple, IMHO.

Reply:
this is fine I think, we can just use newer gcc versions... it's just the compiler.

Reply:
I understand what you mean now, after I tried it.

this would require RHEL8 / CENTOS8, or a newer Kernel. 

it would limit where this would be available to run. So, we need to keep it simple.

Reply:
Exactly. atomic_thread_fence is provided by stdatomic which, in turn, is only available on newer GCC. To work-around that, @franz1981 can probably do some thing like this:

1. Add a check for stdatomic.h on the CMakeLists.txt and, based on its presence, define a HAVE_STDATOMIC_H flag:

```
include (CheckIncludeFiles)
if (UNIX)
	CHECK_INCLUDE_FILES(stdatomic.h HAVE_STDATOMIC_H)
endif (UNIX)
```

2. Define atomic_thread_fence as the previous memory barrier to retain the current behaviour:

```
#if defined(HAVE_STDATOMIC_H)
 #include <stdatomic.h>
#else
 #define memory_order_acquire ""
 #define memory_order_release ""

 #define atomic_thread_fence(x) __asm__ __volatile__ ("":::"memory")
#endif
```
This should make the code build on older versions just like the current one.